pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -246,9 +246,13 @@ func (m *ManagerImpl) GetWatcherHandler() watcher.PluginHandler {
 }
 
 // ValidatePlugin validates a plugin if the version is correct and the name has the format of an extended resource
-func (m *ManagerImpl) ValidatePlugin(pluginName string, endpoint string, versions []string) error {
+func (m *ManagerImpl) ValidatePlugin(pluginName string, endpoint string, versions []string, foundInDeprecatedDir bool) error {
 	klog.V(2).Infof("Got Plugin %s at endpoint %s with versions %v", pluginName, endpoint, versions)
 
+	if foundInDeprecatedDir {
+		return fmt.Errorf("device plugin socket was found in a directory that is no longer supported")
```

Comment:
Why fail with device plugins?

Reply:
Per https://github.com/kubernetes/kubernetes/pull/70494#issuecomment-439509841 the plugin directory is changing, and the old one will be deprecated. That PR moved all consumers (CSI and device plugin to the new dir). This PR makes an exception for CSI 0.x but does not change anything else (meaning device plugin and CSI 1.x drivers are still not permitted to use the old dir).